src/sagas/networkSettings.js

```diff
@@ -1,22 +1,49 @@
-import { all, takeEvery, put, call, race, take, delay, select } from 'redux-saga/effects';
+import { all, takeEvery, put, call, race, delay, select } from 'redux-saga/effects';
 import { config } from '@hathor/wallet-lib';
 import { isEmpty } from 'lodash';
-import AsyncStorage from '@react-native-async-storage/async-storage';
 import { t } from 'ttag';
-import { featureToggleUpdate, networkSettingsUpdateErrors, networkSettingsUpdateFailure, networkSettingsUpdateReady, networkSettingsUpdateSuccess, reloadWalletRequested, types } from '../actions';
-import { HTTP_REQUEST_TIMEOUT, NETWORK, networkSettingsKeyMap, NETWORK_TESTNET, STAGE, STAGE_DEV_PRIVNET, STAGE_TESTNET, WALLET_SERVICE_REQUEST_TIMEOUT } from '../constants';
-import { getFullnodeNetwork, getWalletServiceNetwork } from './helpers';
+import {
+  networkSettingsPersistStore,
+  networkSettingsUpdateInvalid,
+  networkSettingsUpdateFailure,
+  networkSettingsUpdateState,
+  networkSettingsUpdateSuccess,
+  networkSettingsUpdateWaiting,
+  types,
+  reloadWalletRequested,
+  onExceptionCaptured,
+  networkSettingsUpdateReady
+} from '../actions';
+import {
+  NETWORK,
+  networkSettingsKeyMap,
+  NETWORKSETTINGS_STATUS,
+  NETWORK_TESTNET,
+  STAGE,
+  STAGE_DEV_PRIVNET,
+  STAGE_TESTNET,
+  WALLET_SERVICE_REQUEST_TIMEOUT
+} from '../constants';
+import {
+  getFullnodeNetwork,
+  getWalletServiceNetwork,
+} from './helpers';
 import { STORE } from '../store';
 
 /**
- * Initialize network settings saga.
- *
- * It looks up a stored network settings to update the redux state.
+ * Initialize the network settings saga when the wallet starts successfully.
  */
 export function* initNetworkSettings() {
   const customNetwork = STORE.getItem(networkSettingsKeyMap.networkSettings);
   if (customNetwork) {
-    yield put(networkSettingsUpdateSuccess(customNetwork));
+    yield put(networkSettingsUpdateState(customNetwork));
+  }
+
+  const status = yield select((state) => state.networkSettingsStatus);
+  if (status === NETWORKSETTINGS_STATUS.WAITING) {
+    yield put(networkSettingsUpdateSuccess());
+  } else {
+    yield put(networkSettingsUpdateReady());
```

Comment:
I believe this deserves a comment explaining these state changes and why they happen here

Reply:
Added: https://github.com/HathorNetwork/hathor-wallet-mobile/pull/360/commits/3825b2e3978e12bc1900b0797d7f6c7d08f6aa95